hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,82 +10,95 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+Bypass mode can be completely disabled in the {{< regref "CONF" >}} register.
+
+Once the initial bypass mode phase has completed, the ENTROPY_SRC block will switch to FIPS compliant mode. 
+In this mode, once the raw entropy has been health checked, it will be passed into a conditioner block.
+This block will compress the bits such that the entropy bits/physical bits, or min-entropy value, should be improved over the raw data source min-entropy value.
+The compression operation, by default, will be 832 bits out of 1024 tested bits to be compressed into 384 bits.
+
+Health checks are done on the input raw data from the digital noise source.
+There are four health tests that will be performed: repetitive count, adaptive proportion, bucket, and markov tests.
+Each test has a pair of threshold values that determine that pass/fail of the test, one threshold for bypass mode, and one for normal mode.
+By default, all tests are enabled, but can be turn off in the {{< regref "CONF" >}} register. 
+Because of the variability of the noise source, there are several registers that log statistics associated with the health tests. 
+For example, the adaptive proportion test has a high watermark register that logs the highest measured number of ones.
+The {{< regref "ADAPTP_HI_WATERMARKS" >}} register has an entry for both normal and bypass modes.
+This register allows for determining how close the threshold value should be set to the fail over value.
+Specific to the adaptive proportion test, there is also the {{< regref "ADAPTP_LO_WATERMARKS" >}} register, which will hold the lowest number of ones measured.
+To help understand how well the thresholds work through time, a running count of test fails is kept in the {{< regref "ADAPTP_HI_TOTAL_FAILS" >}} register.
+The above example for the adaptive proportion test also applies to the other health tests, with the exception of the low watermark registers.
+See the timing diagrams below for more details on how the health tests work.
+
+The {{< regref "ALERT_THRESHOLD" >}} register determines how many fails can occur before an alert is issued. 
+By default, the current threshold is set to two, such that the occurrence of two failing test cycles back-to-back would provide a very statistically low alpha value.
+The {{< regref "ALERT_FAIL_COUNTS" >}} register holds the total number of fails, plus all of the individual contributing failing tests.
+Setting the {{< regref "ALERT_THRESHOLD" >}} register to zero will disable alert generation.
+
+Firmware has a path to read entropy from the ENTROPY_SRC block.
+The {{< regref "ENTROPY_CONTROL" >}} register allows firmware to set the internal multiplexers to steer entropy data to the {{< regref "ENTROPY_DATA" >}} register.
+The control bit `ES_TYPE` sets whether the entropy will come from the conditioning block or be sourced through the bypass path.
```

Comment:
sorry i probably missed a few discussions here. But will it still be possible to close the firmware path if we want?

Reply:
Yes - there is an efuse that does this. We have a way to read entropy by SW, and a way to close it off. Ziv would like a way to read, modify,and write back to the flow. This is an item in the feature list.

Reply:
sounds good.